tests/lapi/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+include(BuildLuzer)
+include(MakeLuaPath)
+
+if(NOT LUA_EXECUTABLE)
+  message(FATAL_ERROR "${LUA_EXECUTABLE} is not found.")
+endif()
+
+make_lua_path(LUA_CPATH
+  PATHS
+  ${LUZER_LUA_CPATH}
+)
+
+make_lua_path(LUA_PATH
+  PATHS
+  ${LUZER_LUA_PATH}
+  ${CMAKE_CURRENT_SOURCE_DIR}/?.lua
+)
+
+set(DEFAULT_RUNS_NUMBER 5)
+
+string(JOIN " " LIBFUZZER_OPTS
+  -print_final_stats=1
+  -print_pcs=1
+  -reduce_inputs=1
+  -reload=1
+  -report_slow_units=5
+  -runs=$\{RUNS:-${DEFAULT_RUNS_NUMBER}\}
+)
+
+function(create_test)
+  cmake_parse_arguments(
+    FUZZ
+    ""
+    "FILENAME"
+    ""
+    ${ARGN}
+  )
+  get_filename_component(test_name ${FUZZ_FILENAME} NAME_WE)
+  string(REPLACE "_test" "" test_prefix ${test_name})
+  set(dict_path ${PROJECT_SOURCE_DIR}/corpus/${test_prefix}.dict)
+  set(corpus_path ${PROJECT_SOURCE_DIR}/corpus/${test_prefix})
+  if (EXISTS ${dict_path})
+    set(LIBFUZZER_OPTS "${LIBFUZZER_OPTS} -dict=${dict_path}")
+  endif ()
+  if (EXISTS ${corpus_path})
+    set(LIBFUZZER_OPTS "${LIBFUZZER_OPTS} ${corpus_path}")
+  endif ()
+  add_test(NAME ${test_name}
+    COMMAND ${SHELL} -c "${LUA_EXECUTABLE} ${FUZZ_FILENAME} ${LIBFUZZER_OPTS}"
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  )
+  set_tests_properties(${test_name} PROPERTIES
+    LABELS "lapi"
+    ENVIRONMENT "LUA_PATH=${LUA_PATH};LUA_CPATH=${LUA_CPATH};ASAN_OPTIONS=detect_odr_violation=0"
+    DEPENDS ${LUA_EXECUTABLE} ${LUZER_LIBRARY}
+  )
+endfunction()
+
+message(STATUS "Add Lua API test suite")
+file(GLOB tests LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR} *_test.lua)
```

Comment:
It is better to use `${CMAKE_CURRENT_SOURCE_DIR}/*_test.lua` instead, see tarantool/tarantool@7fb8921b4f64dc999ff2df9a047766bb6c653f9b.
Also, I suggest to use the following tests directory structure:

```
$ ls tests/lib
base
coroutine
ffi
math
string
table
```

It group the tests by the library we are testing.

Reply:
Just a suggestion, feel free to ignore (it repeats the name of the directory in the names of the tests, though).